src/components/SearchForm/SearchForm.jsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useHistory } from "react-router-dom";
+import PropTypes from "prop-types";
+import { Select } from "../Select/Select";
+import { DataList } from "../DataList/DataList";
+import { Input } from "../Input/Input";
+import { SERVICES, CATEGORIES, CITIES, PETS } from "../../constants/options";
+import styles from "./search_form.module.scss";
+import { Button } from "../Button/Button";
+import searchButtonIcon from "../../assets/image/searchButton.png";
+
+export const SearchForm = ({ filterAnnouncements }) => {
+  const [filterState, setFilterState] = useState([]);
+  const history = useHistory();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const onSubmit = (data) => {
+    const params = new URLSearchParams();
+    params.set("phrase", data.Phrase);
+    params.set("category", data.Category);
+    params.set("city", data.City);
+    if (data.Animals) {
+      data.Animals.map((animal) => {
```

Comment:
Why use `map` when you don't transform here anything? If you just want to iterate and `append` use `forEach` instead.